core-web/libs/edit-content/src/lib/feature/edit-content/store/features/workflow.feature.ts

```diff
@@ -48,24 +83,115 @@ const initialState: WorkflowState = {
  */
 export function withWorkflow() {
     return signalStoreFeature(
-        {
-            state: type<EditContentState>()
-        },
-        withState(initialState),
-        withComputed(({ workflow }) => ({
-            isLoadingWorkflow: computed(() => workflow.status() === ComponentStatus.LOADING)
+        { state: type<EditContentRootState & ContentState>() },
+        withState(workflowInitialState),
+        withComputed((store) => ({
+            /**
+             * Computed property that determines if the workflow component is in a loading state
+             *
+             * @returns {boolean} True if the workflow component is in a loading state, false otherwise
+             */
+            isLoadingWorkflow: computed(() => store.workflow().status === ComponentStatus.LOADING),
+
+            /**
+             * Gets the workflow scheme for the currently selected scheme ID
+             *
+             * @returns {DotCMSWorkflow | undefined} The workflow scheme if found, undefined otherwise
+             */
+            getScheme: computed<DotCMSWorkflow | undefined>(() => {
+                const currentSchemeId = store.currentSchemeId();
+
+                return currentSchemeId ? store.schemes()[currentSchemeId]?.scheme : undefined;
+            }),
+
+            /**
+             * Computed property that determines if workflow action buttons should be shown.
+             */
+            showWorkflowActions: computed(() =>
+                shouldShowWorkflowActions(
+                    store.schemes(),
+                    store.contentlet(),
+                    store.currentSchemeId()
+                )
+            ),
+
+            /**
+             * Computed property that determines if the workflow selection warning should be shown.
+             * Shows warning when content is new AND no workflow scheme has been selected yet.
+             *
+             * @returns {boolean} True if warning should be shown, false otherwise
+             */
+            showSelectWorkflowWarning: computed(() =>
+                shouldShowWorkflowWarning(
+                    store.schemes(),
+                    store.contentlet(),
+                    store.currentSchemeId()
+                )
+            ),
+
+            /**
+             * Computed property that retrieves the actions for the current workflow scheme.
+             *
+             * @returns {DotCMSWorkflowAction[]} The actions for the current workflow scheme.
+             */
+            getActions: computed(() =>
+                getWorkflowActions(
+                    store.schemes(),
+                    store.contentlet(),
+                    store.currentSchemeId(),
+                    store.currentContentActions()
+                )
```

Comment:
maybe these functions can recieve one object, instead of 4 params

Reply:
yes. good idea.

Reply:
changed. All of them.